docs/reference/snapshot-restore/restore-snapshot.asciidoc

```diff
@@ -29,23 +25,57 @@ A snapshot can be restored using the following command:
 POST /_snapshot/my_backup/snapshot_1/_restore
 -----------------------------------
 
-By default, all indices in the snapshot are restored, and the cluster state is
-*not* restored. It's possible to select indices that should be restored as well
+By default, all data streams and indices in the snapshot are restored, but the cluster state is
+*not* restored. It's possible to select specific data streams or indices that should be restored as well
 as to allow the global cluster state from being restored by using `indices` and
-`include_global_state` options in the restore request body. The list of indices
-supports <<multi-index,multi index syntax>>. The `rename_pattern`
-and `rename_replacement` options can be also used to rename indices on restore
+`include_global_state` options in the restore request body. The list
+supports <<multi-index,multi index syntax>>.
+
+[WARNING]
+====
+Each data stream requires an associated
+<<create-a-data-stream-template,composable template>>. The stream uses this
+template to create new backing indices.
+
+When restoring a data stream, ensure a matching template exists for the stream.
+You can do this using one of the following methods:
+
+* Check for existing templates that match the stream. If no matching template
+  exists, <<create-a-data-stream-template,create one>>.
+
+* Restore a global cluster state that includes a matching template for the
+  stream.
+
+If no composable template matches a data stream, the stream cannot
+<<manually-roll-over-a-data-stream,roll over>> or create new backing indices.
+====
+
+The `rename_pattern`
+and `rename_replacement` options can be also used to rename data streams and indices on restore
 using regular expression that supports referencing the original text as
 explained
 http://docs.oracle.com/javase/6/docs/api/java/util/regex/Matcher.html#appendReplacement(java.lang.StringBuffer,%20java.lang.String)[here].
+
+If you rename a restored data stream, its backing indices are also
+renamed. For example, if you rename the `logs` data stream to `restored-log`,
+the backing index `.ds-logs-000005` is renamed to `.ds-restored-logs-000005`.
```

Comment:
Nit: It's either both singular `restored-log` and `.ds-restored-log-000005` or both plural.

Reply:
Thanks for catching this. Fixed.